### PR TITLE
BRAYNS 488 - Add engine tests

### PR DESCRIPTION
**Pull request by NadirRoGue**

### CODE REVIEW

File: brayns/engine/common/SimulationScanner.cpp [hunk 1]
```diff
@@ -40,16 +44,29 @@ struct SimulationUpdater
         currentFrame = std::min(endFrame, std::max(startFrame, currentFrame));
```
Comment (Adrien4193) [inline brayns/engine/common/SimulationScanner.cpp:44]: std::clamp ?

Comment (Adrien4193) [reply brayns/engine/common/SimulationScanner.cpp:44]: Forget it

File: tests/helpers/ImageValidator.h [hunk 2]
```diff
@@ -83,7 +83,7 @@ class ImageValidator
 
     static bool _isSavePngEnabled()
     {
-        return std::getenv("BRAYNS_TEST_SAVE_PNG");
+        return true; // std::getenv("BRAYNS_TEST_SAVE_PNG");
```
Comment (Adrien4193) [inline tests/helpers/ImageValidator.h:86]: Is it intentional to leave it as true ?

File: tests/unit/brayns/engine/TestSimulationScanner.cpp [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        models.add(std::move(model));
+    }
+};
+}
+
+TEST_CASE("simulation_scanner")
+{
+    auto params = brayns::ParametersManager(0, nullptr);
+    auto engine = brayns::Engine(params);
+    (void)engine;
```
Comment (Adrien4193) [inline tests/unit/brayns/engine/TestSimulationScanner.cpp:51]: PlaceholderEngine ?

File: tests/helpers/ImageValidator.h (hunk 2 above)
Comment (NadirRoGue) [reply tests/helpers/ImageValidator.h]: Fixed

File: tests/unit/brayns/engine/TestSimulationScanner.cpp (hunk 3 above)
Comment (NadirRoGue) [reply tests/unit/brayns/engine/TestSimulationScanner.cpp:51]: It initializes the ospray device, needed when commiting objects

**Review by Adrien4193 — APPROVED** [review]

**Pull request merged**